forms-flow-web/src/components/Form/List.js

```diff
@@ -201,11 +207,15 @@ const List = React.memo((props) => {
     dispatch(setBPMFormListPage(newState.page));
   };
   const handleSearch = () => {
-    if (searchText != searchInputBox.current.value) {
+    if (searchText != searchInputBox.current.value && isSearchValid) {
       dispatch(setBPMFormListPage(1));
 
       dispatch(setBpmFormSearch(searchInputBox.current.value));
     }
+      else {
+        searchInputBox.current.value !== '' ? toast.error("Please remove the special charactors...! ") : '';
```

Comment:
same here. You only need to check if the validator is allowing the search and notify the same. Please avoid the Spaghetti code.